samples/subsys/fs/littlefs_blk/src/main.c

```diff
@@ -0,0 +1,272 @@
+#define MAX_PATH_LEN 255
+#define TEST_FILE_SIZE 547
+
+static int lsdir(const char *path);
+struct fs_littlefs lfsfs;
+
+static uint8_t file_test_pattern[TEST_FILE_SIZE];
+
+static void incr_pattern(uint8_t *p, uint16_t size, uint8_t inc)
+{
+	int i;
+
+	for (i = 0; i < size; i += 8) {
+		p[i] += inc;
+	}
+
+	p[size - 1] += inc;
+}
+
+static void init_pattern(uint8_t *p, uint16_t size)
+{
+	uint8_t v = 0x1;
+	int i;
+
+	memset(p, 0x55, size);
+
+	for (i = 0; i < size; i += 8) {
+		p[i] = v++;
+	}
+
+	p[size - 1] = 0xAA;
+}
+
+static void print_pattern(uint8_t *p, uint16_t size)
+{
+	int i, j = size / 16, k;
+
+	for (k = 0, i = 0; k < j; i += 16, k++) {
+		LOG_PRINTK("%02x %02x %02x %02x %02x %02x %02x %02x ",
+		       p[i], p[i+1], p[i+2], p[i+3],
+		       p[i+4], p[i+5], p[i+6], p[i+7]);
+		LOG_PRINTK("%02x %02x %02x %02x %02x %02x %02x %02x\n",
+		       p[i+8], p[i+9], p[i+10], p[i+11],
+		       p[i+12], p[i+13], p[i+14], p[i+15]);
+
+		/* Mark 512B (sector) chunks of the test file */
+		if ((k + 1) % 32 == 0) {
+			LOG_PRINTK("\n");
+		}
+	}
+
+	for (; i < size; i++) {
+		LOG_PRINTK("%02x ", p[i]);
+	}
+
+	LOG_PRINTK("\n");
+}
+
+
+static int littlefs_binary_file_adj(char *fname)
+{
+	struct fs_dirent dirent;
+	struct fs_file_t file;
+	int rc;
+
+	/*
+	 * Uncomment below line to force re-creation of the test pattern
+	 * file on the littlefs FS.
+	 */
+	/* fs_unlink(fname); */
+	fs_file_t_init(&file);
+
+	rc = fs_open(&file, fname, FS_O_CREATE | FS_O_RDWR);
+	if (rc < 0) {
+		LOG_ERR("FAIL: open %s: %d", log_strdup(fname), rc);
+		return rc;
+	}
+
+	rc = fs_stat(fname, &dirent);
+	if (rc < 0) {
+		LOG_ERR("FAIL: stat %s: %d", log_strdup(fname), rc);
+		return rc;
+	}
+
+	/* Check if the file exists - if not just write the pattern */
+	if (rc == 0 && dirent.type == FS_DIR_ENTRY_FILE && dirent.size == 0) {
+		LOG_INF("Test file: %s not found, create one!",
+			log_strdup(fname));
+		init_pattern(file_test_pattern, sizeof(file_test_pattern));
+	} else {
+		rc = fs_read(&file, file_test_pattern,
+			     sizeof(file_test_pattern));
+		if (rc < 0) {
+			LOG_ERR("FAIL: read %s: [rd:%d]",
+				log_strdup(fname), rc);
+			return rc;
+		}
+		incr_pattern(file_test_pattern, sizeof(file_test_pattern), 0x1);
+	}
+
+	LOG_PRINTK("------ FILE: %s ------\n", fname);
+	print_pattern(file_test_pattern, sizeof(file_test_pattern));
+
+	rc = fs_seek(&file, 0, FS_SEEK_SET);
+	if (rc < 0) {
+		LOG_ERR("FAIL: seek %s: %d", log_strdup(fname), rc);
+		return rc;
+	}
+
+	rc = fs_write(&file, file_test_pattern, sizeof(file_test_pattern));
+	if (rc < 0) {
+		LOG_ERR("FAIL: write %s: %d", log_strdup(fname), rc);
+		return rc;
+	}
+
+	rc = fs_close(&file);
+	if (rc < 0) {
+		LOG_ERR("FAIL: close %s: %d", log_strdup(fname), rc);
+		return rc;
+	}
+
+	return 0;
+}
+
+static int littlefs_increase_infile_value(char *fname)
+{
+	uint8_t boot_count = 0;
+	struct fs_file_t file;
+	int rc;
+
+	fs_file_t_init(&file);
+	rc = fs_open(&file, fname, FS_O_CREATE | FS_O_RDWR);
```

Comment:
@nvlsianpu  @de-nordic One question, I try to test it on the boards with SDMMC support I have, but it does not create a file for me, any ideas?

Reply:
> @lmajewski since GitHub tends to make it difficult to see older threads, here is the [comment](https://github.com/zephyrproject-rtos/zephyr/pull/40704/commits/6fc45e4b38afa278f310f05ffe902430de0f75e3#r767633571) that needs addressing the most urgently: Why do we need a new sample, given that an existing one (`samples/subsys/fs/littlefs`) already exists? Can't we just add an extra overlay and `.conf` file to the existing sample? As well of course as adapting the source code so that it supports littlefs on both flash and block backends.

@carlescufi  IMHO, New test file can address issues which are problematic for block devices (the LBA sized blocks,
which are not corresponding to flash erase block size nor the read data chunk size). 

What is the problem with running `littlefs` and then `littlefsblk` ? In my opinion it is better to have many small tests than one large ....

Reply:
> @nvlsianpu @de-nordic One question, I try to test it on the boards with SDMMC support I have, but it does not create a file for me, any ideas?

@jfischer-no - Please double check if the `littlefs` is correctly created on the SD card. It works flawlessly on Debian 10, but there were some issues on recent Ubuntu.

Reply:
SD card is formated by the littlefs support in Zephyr, `fs_open(&file, fname, FS_O_CREATE | FS_O_RDWR);`  fails in this case.